File: A_Catch_the_Coin.py
def catch_the_coin(n, coins):
    for x, y in coins:
        # Check if Monocarp can reach before the coin falls
        if y < -1:
            print("NO")
        else:
            print("YES")

# Input
n = int(input())
coins = [tuple(map(int, input().split())) for _ in range(n)]

# Output result
catch_the_coin(n, coins)

File: A_Catch_the_Coin.java
import java.util.Scanner;

public class A_Catch_the_Coin {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();

        for (int i = 0; i < n; i++) {
            int x = scanner.nextInt();
            int y = scanner.nextInt();

            // Check if Monocarp can reach before the coin falls
            if (y < -1) {
                System.out.println("NO");
            } else {
                System.out.println("YES");
            }
        }

        scanner.close();
    }
}
